pennylane/dla/lie_closure.py

```diff
@@ -115,7 +247,7 @@ def __init__(self, generators):
         self._basis = []
         rank = 0
 
-        self._M = np.zeros((num_pw, rank), dtype=float)
+        self._M = np.zeros((num_pw, rank), dtype=complex)
```

Comment:
Just out of curiousity: Why do we need this?

Reply:
Good question 🤔  no Idea why I changed this in this PR; but all tests seem to pass when changing it back

Reply:
Gonna revert that change 👍 ✅